app/models/components/course/experience_points_records_ability_component.rb

```diff
@@ -0,0 +1,54 @@
+module Course::ExperiencePointsRecordsAbilityComponent
+  include AbilityHost::Component
+
+  def define_permissions
+    # If administrators are not excluded, cancancan will throw an exception:
+    # "Unable to merge an Active Record scope with other conditions..."
+    if user && !user.administrator?
+      allow_staff_manage_experience_points_records
+      allow_staff_and_points_owner_read_experience_points_records
+    end
+
+    super
+  end
+
+  private
+
+  def allow_staff_and_points_owner_read_experience_points_records
```

Comment:
Assignment Branch Condition size for allow_staff_and_points_owner_read_experience_points_records is too high. [22.18/15]